tests/li_wand_2016/test_loss.py

```diff
@@ -1,29 +1,188 @@
+            op = loss.LiWand2016MRFOperator(encoder, 3, impl_params=impl_params,)
+            op.set_target_image(target_image)
+            actual = op(input_image)
+
+            if impl_params:
+                target_repr = loss.extract_normalized_patches2d(
+                    target_enc, patch_size, stride
+                )
+                input_repr = loss.extract_normalized_patches2d(
+                    input_enc, patch_size, stride
+                )
+            else:
+                target_repr = extract_patches2d(target_enc, patch_size, stride)
+                input_repr = extract_patches2d(input_enc, patch_size, stride)
+
+            score = mrf_loss(input_repr, target_repr, reduction=loss_reduction)
+            desired = score * score_correction_factor
+
+            assert actual == ptu.approx(desired)
+
+
+def test_li_wand_2016_style_loss(subtests):
+    style_loss = loss.li_wand_2016_style_loss()
+    assert isinstance(style_loss, ops.MultiLayerEncodingOperator)
+
+    with subtests.test("encoding_ops"):
+        assert all(isinstance(op, ops.MRFOperator) for op in style_loss.operators())
+
+    configs = ((True, 1e-4, 2, 1, 5e-2, 1, 7.5), (False, 1e0, 1, 3, 5e-2, 2, 7.5))
+    for (
+        impl_params,
+        score_weight,
+        stride,
+        num_scale_steps,
+        scale_step_width,
+        num_rotate_steps,
+        rotate_step_width,
+    ) in configs:
+        style_loss = loss.li_wand_2016_style_loss(impl_params=impl_params)
+        layers, layer_weights, op_stride, op_target_transforms = zip(
+            *[
+                (op.encoder.layer, op.score_weight, op.stride, op.target_transforms)
+                for op in style_loss.operators()
+            ]
+        )
+
+        with subtests.test("layers"):
+            assert set(layers) == {"relu3_1", "relu4_1"}
+
+        with subtests.test("stride"):
+            assert op_stride == (to_2d_arg(stride),) * len(layers)
+
+        with subtests.test("score_weight"):
+            assert style_loss.score_weight == pytest.approx(score_weight)
+
+        with subtests.test("layer_weights"):
+            assert layer_weights == (1.0,) * len(layers)
+
+        scaling_factors = np.arange(
+            -num_scale_steps, num_scale_steps + 1, dtype=np.float
+        )
+        scaling_factors = 1.0 + (scaling_factors * scale_step_width)
+
+        rotation_angles = np.arange(
+            -num_rotate_steps, num_rotate_steps + 1, dtype=np.float
+        )
+        rotation_angles *= rotate_step_width
+
+        for i, values in enumerate(itertools.product(scaling_factors, rotation_angles)):
+            scaling_factor, rotation_angle = values
+            with subtests.test("scaling_factor"):
+                assert op_target_transforms[0][i].scaling_factor == scaling_factor
+
+            with subtests.test("rotation_angles"):
+                assert op_target_transforms[0][i].rotation_angle == rotation_angle
```

Comment:
Not sure about that. I think we should mock `MRFOperator.scale_and_rotate_transforms` and check if its called with the correct parameters.

Reply:
I think you are right, here you should not check if the transformations are correct, but if this function is called correctly. The actual test should take place in the tests of the `MRFOperator`.